### PR TITLE
Getter Enums

### PR DESCRIPTION
**Pull request by miguelangel-dev**
## Goal

Generate Getter for commons associated values in Enums.

## Implementation details

A new visitor is created to visit commons `associated values` in enums and create an extension to add a static computed property containing the Getter for each one of them.

A new generator is created using this visitor and passing a default set of imports (`BowOptics`, `Bow` need to be imported in the generated code).

## Testing details

Generated code is tested using Snapshot testing and the generated output is put back into the project to guarantee its compilation.

### CODE REVIEW

File: Tests/MetaTests/Generators/GetterEnumGeneratorTests.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import XCTest
+import SnapshotTesting
+import Meta
+
+class GetterEnumGeneratorTests: XCTestCase {
+    func testGetterEnumGeneratorMultipleFile() {
+        assertSnapshot(matching: URL.meta.fixtures, as: .generatedCode(using: GetterEnumGenerator()))
```
Comment (truizlop) [inline Tests/MetaTests/Generators/GetterEnumGeneratorTests.swift:7]: What if the associated values appear in different order in each case?

Comment (truizlop) [reply Tests/MetaTests/Generators/GetterEnumGeneratorTests.swift:7]: Can you add a test case for that?

Comment (miguelangel-dev) [reply Tests/MetaTests/Generators/GetterEnumGeneratorTests.swift:7]: In `GetterEnumVisitor:commonAssociatedFields` you can find it:
```swift
return cases.first?.associatedValues.filter { field in commonsFields.contains(field) } ?? []
```
it keeds the same order in the return type

Comment (miguelangel-dev) [reply Tests/MetaTests/Generators/GetterEnumGeneratorTests.swift:7]: Add test to verify it renders always the same result

**Review by truizlop — APPROVED** [review]

**Pull request merged**